File: c++/study/getcwd_example.c
/*******************************************************************************
* 文 件 名: getcwd_example.c
* 文件描述: getcwd函数的使用
* 作    者: HanKin
* 创建日期: 2023.10.17
* 修改日期：2023.10.17
*
* Copyright (c) 2023 HanKin. All rights reserved.
*******************************************************************************/
#include <stdio.h>
#include <unistd.h>

int main()
{
    char cwd[1024] = { 0 };
    if (getcwd(cwd, sizeof(cwd)) != NULL) {
        printf("当前工作目录: %s\n", cwd);
    } else {
        perror("获取工作目录失败");
        return 1;
    }
    chdir("/path/to/directory");
    return 0;
}

File: python/study/others/getcwd_example.py
# -*- coding: utf-8 -*-
"""
文 件 名: getcwd_example.py
文件描述: 获取当前路径
作    者: HanKin
备    注: 
创建日期: 2025.03.31
修改日期：2025.03.31

Copyright (c) 2025 HanKin. All rights reserved.
"""
import os

print(os.getcwd())
print(os.path.join(os.getcwd(), "build"))

"""
(base) D:\Github\Storage\python\study\others>python getcwd_example.py
D:\Github\Storage\python\study\others
D:\Github\Storage\python\study\others\build
"""